File: samples/CslaNet/vb/DeepData/DeepData.Library/LineItemList.vb
<Serializable()> _
Public Class LineItemList
  Inherits ReadOnlyListBase(Of LineItemList, LineItemInfo)

#Region " Business Methods "

  Public Function FindById(ByVal id As Integer) As LineItemInfo

    For Each child As LineItemInfo In Me
      If child.Id = id Then
        Return child
      End If
    Next
    Return Nothing

  End Function

#End Region

#Region " Factory Methods "

  Friend Shared Function NewList() As LineItemList

    Return New LineItemList

  End Function

  Private Sub New()

  End Sub

#End Region

#Region " Data Access "

  Friend Sub LoadChild(ByVal data As SafeDataReader)

    IsReadOnly = False
    Add(LineItemInfo.GetItem(data))
    IsReadOnly = True

  End Sub

#End Region

End Class

File: samples/CslaNet/cs/DeepData/DeepData.Library/LineItemList.cs
using Csla;
using Csla.Data;

namespace DeepData.Library
{
	[System.Serializable()]
	public partial class LineItemList
	  : ReadOnlyListBase<LineItemList, LineItemInfo>
	{
		#region " Business Methods "

		///<summary>
		///</summary>
		///<param name="id"></param>
		///<returns></returns>
		public LineItemInfo FindById(int id)
		{
			foreach (LineItemInfo child in this)
			{
				if (child.Id == id)
				{
					return child;
				}
			}
			return null;
		}

		#endregion

		#region " Factory Methods "

		internal static LineItemList NewList()
		{
			return new LineItemList();
		}

		private LineItemList()
		{
		}

		#endregion

		#region " Data Access "

		internal void LoadChild(SafeDataReader data)
		{
			IsReadOnly = false;
			Add(LineItemInfo.GetItem(data));
			IsReadOnly = true;
		}

		#endregion

	}
}
